### PR TITLE
Updates required to handle new insights tar.gz format

### LINKED ISSUE #94: Modify qpc to handle new insights report slices
## Feature qpc report insights:
Currently the `qpc report insights` command downloads & unzips the payload returned from the server. Since the insights report is becoming multiple payloads (a metadata payload & one or more report payloads), we want this command to download the tar.gz & not unzip it. 

## Feature qpc insights upload:
QPC insights upload should remove the --json option.  No local upload.

### Is your feature request related to a problem?
Insights reports can be very large. Additionally, we can't validate a report without loading the JSON. The act of loading a very large JSON file could cause our service to have memory issues.

Since we are sending a tar.gz to the file upload service, we should split out the metadata into a small JSON payload that will always load and a report JSON payload (possibly multiple report payloads).

### Describe the solution you'd like
We should no longer unzip the tar.gz. 
___

## Acceptance Criteria:

- [ ] Verify that `qpc report insights` command no longer unzips the tar.gz returned by the server
- [ ] Verify that `qpc insights upload` command no longer has --json option

___

### PR DESCRIPTION
**Pull request by kholdaway**
Closes #94

### CODE REVIEW

File: insights_configuration.rst [hunk 1]
```diff
@@ -62,10 +66,11 @@ Download Insights Core RPM
 ^^^^^^^^^^^^^^^^^^^^^^^^^^
 Download the last stable version of the Insights core::
 
-    curl https://api.access.redhat.com/r/insights/v1/static/core/insights-core.egg.asc > last_stable.egg.asc
-    sudo mv last_stable.egg.asc /var/lib/insights/last_stable.egg.asc
-    curl https://api.access.redhat.com/r/insights/v1/static/core/insights-core.egg > last_stable.egg
-    sudo mv last_stable.egg /var/lib/insights/last_stable.egg
+sudo sh lay-the-eggs-osx.sh
```
Comment (myersCody) [inline insights_configuration.rst]: I think this needs to be tabbed over to be shown as code in the rst.

Comment (kholdaway) [reply insights_configuration.rst]: Fixed.  Thanks.

File: qpc/insights/upload.py [hunk 2]
```diff
@@ -95,8 +87,8 @@ def __init__(self, subparsers):
         input_group.add_argument('--scan-job', dest='scan_job_id',
                                  metavar='SCAN_JOB_ID',
                                  help=_(messages.INSIGHTS_SCAN_JOB_ID_HELP))
-        input_group.add_argument('--json-file', dest='json_file', metavar='JSON_FILE',
-                                 help=_(messages.INSIGHTS_INPUT_JSON_HELP))
+        input_group.add_argument('--gzip-file', dest='gzip_file', metavar='GZIP_FILE',
```
[the remark this answers is not in the document]
Comment (kholdaway) [reply qpc/insights/upload.py]: yes.

[the remark this answers is not in the document]
Comment (kholdaway) [reply qpc/insights/upload.py]: what should we call it for input?

File: qpc/insights/upload.py [hunk 3]
```diff
@@ -164,42 +156,19 @@ def _validate_args(self):
         print(_(messages.INSIGHTS_IS_VERIFIED))
 
         # obtaining the report as tar.gz
-        if self.args.json_file:
+        if self.args.gzip_file:
```
[the remark this answers is not in the document]
Comment (kholdaway) [reply qpc/insights/upload.py]: Cool, I will use that.

File: qpc/insights/upload.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (abaiken) [reply qpc/insights/upload.py]: `input-file` or we could change them both to say `gzip-file` ?

[the remark this answers is not in the document]
Comment (kholdaway) [reply qpc/insights/upload.py]: let's go with input-file

File: qpc/insights/upload.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (kholdaway) [reply qpc/insights/upload.py]: I decided to just check for the suffix since it was one line of code.

File: docs/source/man.rst [hunk 4]
```diff
@@ -642,6 +642,10 @@ The ``qpc insights upload`` command can be used to upload an insights report to
 
   Contains the report identifier to use to retrieve and upload the insights report. Mutually exclusive with the ``--scan-job`` option.
 
+``--input-file=path to tar.gz containing the Insights report``
```
Comment (abaiken) [inline docs/source/man.rst]: you need to add the input file to the top level `qpc insights upload command`

Comment (kholdaway) [reply docs/source/man.rst]: Done.  Good catch.

**Pull request merged**